test/libsolidity/smtCheckerTests/special/msg_value_inheritance_3.sol

```diff
@@ -0,0 +1,22 @@
+pragma experimental SMTChecker;
+
+contract A {
+	uint public x = msg.value;
+	constructor() {
+		assert(x == 0); // can fail when A is constructed as part of C
+	}
+}
+
+contract B {
+	constructor() payable {
+		assert(msg.value >= 0); // should hold
+	}
+}
+contract C is A, B {
+	constructor() A() B() payable {
+		assert(msg.value >= 0); // should hold
+	}
+}
```

Comment:
Just curious what happens with the following

```
pragma experimental SMTChecker;

contract A {
	uint public x = msg.value;
	constructor() {
		assert(x == 0); // can fail when A is constructed as part of C or B
	}
}

contract B is A {
	constructor() A() payable {
		assert(msg.value >= 0); // should hold
	}
}
contract C is B, A {
	constructor() B() A() payable {
		assert(msg.value >= 0); // should hold
	}
}
```

and 

```
pragma experimental SMTChecker;

contract A {
	constructor() payable {
		assert(x == 0); // should hold when constructed via C, fail otherwise?
	}
}

contract B is A {
	constructor() A() payable {
		assert(msg.value >= 0); // should hold
	}
}
contract C is A, B {
	constructor() A() B() {
		assert(msg.value == 0); // should hold
	}
}
```

Reply:
@bshastry , @leonardoalt , I have split the function `txConstraints` to `txTypeConstraints` and `txFunctionConsttraints`, I think it makes sense.
This also showed that previously the `non-payable` constraint `msg.value == 0` was missing for `fallback` function. 
It also forms the conjunction of the constraints in a slightly different way, so there are a bunch of changes in command line tests where the BMC formula is printed to the output.

@bshastry , I am not sure if I have addressed all your comments, let me know I missed something.

Reply:
@blishko ah I should've tried those test cases before typing them out. Here you go

```
pragma experimental SMTChecker;

contract A {
	uint public x = msg.value;
	constructor() {
		assert(x == 0); // can fail when A is constructed as part of C or B
	}
}

contract B is A {
	constructor() A() payable {
		assert(msg.value >= 0); // should hold
	}
}
contract C is A, B {
	constructor() B() payable {
		assert(msg.value >= 0); // should hold
	}
}
```

```
pragma experimental SMTChecker;

contract A {
        uint x;
	constructor() payable {
		assert(x == 0); // should hold when constructed via C, fail otherwise?
	}
}

contract B is A {
	constructor() A() payable {
		assert(msg.value >= 0); // should hold
	}
}
contract C is A, B {
	uint y = msg.value;
	constructor() B() {
		assert(y == 0); // should hold
	}
}
```

Reply:
Regarding the first example, it is as you write in the comments. The assertion in `A` can fail when constructed as partt of `C` or `B`. The other assertions hold. I think this is the same situation as in the test case `msg_value_inheritance_2.sol`.

Regarding the second example, as you have written it, all assertions hold. If you meant `uint x = msg.value` then the assertion in `A` can fail when contract `A` is constructed with value greater than 0, which is possible since constructor of `A` is `payable`.